doc/architecture-decisions/0004-strategy-for-securing-network-traffic.md

```diff
@@ -0,0 +1,47 @@
+* **System Components:** Cloud Foundry core services that implement power the platform.
+For example, UAA, Cloud Controller, Log-Cache, etc.
+
+We expect all of these components to want to communicate with each other in some fashion and we will leverage some existing Istio functionality to accomplish securing internal traffic.
+
+See the following matrix:
+
+| Source\Destination | External                                                                                                                      | Apps                                                                                                                                    | System Components                                                                                                                                                                                                   |
+|--------------------|-------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| External           | N/A                                                                                                                           | Operator provides wildcard certs to Ingress Gateway. Gateway to backend is secured using [Istio Mesh mTLS](#istio-mesh-mtls) | Operator provides wildcard certs for the system domain to Ingress Gateway. Gateway to backend is secured using [Istio Mesh mTLS](#istio-mesh-mtls)                                                                                 |
+| Apps               | Mesh can be leveraged with Istio Destination Rules and Service Entries. [Istio defaults to PERMISSIVE egress](https://istio.io/docs/tasks/traffic-management/egress/egress-control/).                                                       | App-to-app traffic will be denied by default. If enabled, will be secured using [Istio Mesh mTLS](#istio-mesh-mtls) | [Istio Mesh mTLS](#istio-mesh-mtls)<sup>0</sup>                                                                                                                     |
+| System Components  | CAs for external destinations can be provided<sup>1</sup> | [Istio Mesh mTLS](#istio-mesh-mtls)<sup>2</sup> | [Istio Mesh mTLS](#istio-mesh-mtls) |
+
+
+
+### Istio Mesh mTLS
+When we refer to "Istio Mesh mTLS" in the matrix above, we are assuming the following Istio functionality is leveraged:
+
+* [Istio Auto mTLS](https://istio.io/docs/tasks/security/authentication/auto-mtls/)
+* Have a default Istio mesh [policy](https://istio.io/docs/tasks/security/authentication/authn-policy/) enforcing `STRICT` mTLS
```

Comment:
In the Mesh Policy, the [peer authentication method will be set to mtls](https://istio.io/docs/reference/config/security/istio.authentication.v1alpha1/#PeerAuthenticationMethod). That looks like it [defaults to `STRICT`](https://istio.io/docs/reference/config/security/istio.authentication.v1alpha1/#MutualTls), but we might want to explicitly write `STRICT` anyways because we care about that even if the default changes. I'd want us to tweak the wording to say something like "Have a default Istio mesh policy that sets the peer authentication method to mTLS and sets the mTLS mode to STRICT"